File: scratch/crawl.js
const fs = require('fs');

async function main() {
  const res = await fetch('https://www.agrya.in/', {
    headers: { 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7)' }
  });
  const html = await res.text();
  fs.writeFileSync('scratch/home.html', html);

  const linkMatches = [...html.matchAll(/href=["']([^"']+)["']/g)].map(m => m[1]);
  const imgMatches = [...html.matchAll(/src=["']([^"']+)["']/g)].map(m => m[1]);
  const jsonLdMatches = [...html.matchAll(/<script type="application\/ld\+json">([\s\S]*?)<\/script>/gi)].map(m => m[1]);
  const metaMatches = [...html.matchAll(/<meta\s+([^>]+)>/gi)].map(m => m[1]);

  console.log('--- FOUND LINKS (' + linkMatches.length + ') ---');
  console.log([...new Set(linkMatches)].sort());

  console.log('--- FOUND IMAGES/SCRIPTS (' + imgMatches.length + ') ---');
  console.log([...new Set(imgMatches)].sort());

  console.log('--- JSON-LD ---');
  console.log(jsonLdMatches);

  console.log('--- META TAGS ---');
  console.log(metaMatches);
}

main().catch(console.error);

File: scratch/crawl.py
import urllib.request
import re
from bs4 import BeautifulSoup
import json
import os

url = "https://www.agrya.in/"
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
try:
    with urllib.request.urlopen(req) as resp:
        html = resp.read().decode('utf-8', errors='ignore')
except Exception as e:
    print(f"Error fetching {url}: {e}")
    exit(1)

with open("scratch/home.html", "w", encoding="utf-8") as f:
    f.write(html)

soup = BeautifulSoup(html, "html.parser")

# Find all links
links = set()
for a in soup.find_all("a", href=True):
    href = a['href']
    links.add(href)

print("--- FOUND LINKS ---")
for l in sorted(links):
    print(l)

# Find all images
images = set()
for img in soup.find_all("img", src=True):
    images.add(img['src'])

print("\n--- FOUND IMAGES ---")
for img in sorted(images):
    print(img)

# Find JSON-LD
print("\n--- JSON-LD ---")
for s in soup.find_all("script", type="application/ld+json"):
    print(s.string)

# Meta tags
print("\n--- META TAGS ---")
for m in soup.find_all("meta"):
    print(m.attrs)
